From 9ad9cc216af1d69e710bce5c3258a3d7a6620f1e Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Fri, 19 Apr 2013 20:56:23 +0200
Subject: [PATCH 45/65] r92su: use rcu_barrier in module unload path

synchronize_rcu does not wait for the callbacks to finish
like rcu_barrier does.

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 r92su/main.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/wireless/r92su/main.c
+++ b/drivers/net/wireless/r92su/main.c
@@ -1696,6 +1696,7 @@ void r92su_unalloc(struct r92su *r92su)
 		wiphy_unregister(r92su->wdev.wiphy);
 
 	synchronize_rcu();
+	rcu_barrier();
 
 	destroy_workqueue(r92su->wq);
 	mutex_destroy(&r92su->lock);
